src/main/java/module-info.java

```diff
@@ -0,0 +1,8 @@
+/**
+ * This module provides a Java API that allows building User Defined Function scripts for Exasol.
+ */
+module com.exasol {
```

Comment:
Module name is too generic.
```suggestion
module com.exasol.udf {
```